lib/bundler/shared_helpers.rb

```diff
@@ -287,8 +287,11 @@ def set_env(key, value)
     public :set_env
 
     def set_bundle_variables
+      # bundler exe & lib folders have same root folder
       exe_file = File.expand_path("../../../exe/bundle", __FILE__)
-      # for Ruby core repository
+      # bundler is a default gem, exe path is separate
+      exe_file = Bundler.rubygems.bin_path("bundler", "bundle", VERSION) unless File.exist?(exe_file)
+      # for Ruby core repository testing
       exe_file = File.expand_path("../../../../bin/bundle", __FILE__) unless File.exist?(exe_file)
```

Comment:
Isn't this path incorrect though? I look at current's repository structure in ruby-core and it's like

```
|--- bin/bundle
|--- lib/bundler/shared_helpers.rb
```

So, I think one level of `../` is incorrect?

Reply:
Yes, sort of.  It's incorrect from the source folder.

But, we have three folders, source, build, and install.  With ruby-loco, since it is now used in so many repos for testing, I patch the test files and run all testing from install.  I also do a CLI check from install  (I'll add a nested 'bundle exec' test sometime soon).  So I can't really check it from ruby-loco, as it's using the install folder, which finds the file with the addition.

Anyway, long story short, I did a similar PR in Ruby, and tests passed.  So I don't know if
```ruby
File.expand_path("../../../../bin/bundle", __FILE__)
```

is run from the 'build' folder or the 'source' folder.

I left It as is, I could test it in my Ruby fork on Travis & Appveyor?  Am I making sense?

Reply:
Yes, you are. Running the tests from the install folder makes sense to me and it helps catching issues like this one, so thanks.

I'll let @hsbt answer the question about ruby-core testing, but the fix for bundler installed as a default gem looks good to me.

Reply:
I've got a build with a patch with debug output running on Travis now...  20 minutes maybe...

Reply:
@deivid-rodriguez 

Thanks for looking at that, as I ignored the existing code.  Using three levels, it finds the bin file, and it's in the build folder.  So, I pushed that change also.  Maybe core testing doesn't test it...